File: [프로그래머스] 코딩테스트 입문_다항식 더하기.js
function solution(polynomial) {
    let arr = polynomial.split(" + ")
    let x = 0
    let num = 0
    for(let i of arr){
        if(i.includes("x")){
            i === "x" ? x += 1 : x += parseInt(i);
        } else {
            num += parseInt(i);
        }
    }
    if(x > 0 && num > 0){
        return x === 1 ? "x + " + num.toString() : x.toString() + "x + " + num.toString();
    } else if(x == 0){
        return num.toString();
    } else if(num == 0){
        return x === 1 ? "x" : x.toString() + "x";
    }
}

File: [프로그래머스] 코딩테스트 입문_다항식 더하기.py
def solution(polynomial):
    arr = polynomial.replace(" ", "").split("+")
    x, num = 0, 0
    for i in arr:
        if "x" in i:
            if len(i) >= 2:
                x += int(i[:-1])
            else:
                x += 1
        else:
            num += int(i)
    if num > 0 and x > 0:
        if x == 1:
            return "x + "+str(num)
        else:
            return str(x)+"x + "+str(num)
    elif x <= 0:
        return str(num)
    elif num <= 0:
        if x == 1:
            return "x"
        else:
            return str(x)+"x"